tests/HypothesisClient/Credentials/JWTSigningCredentialsTest.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace tests\eLife\HypothesisClient\Credentials;
+
+use eLife\HypothesisClient\Clock\Clock;
+use eLife\HypothesisClient\Credentials\JWTSigningCredentials;
+use Firebase\JWT\JWT;
+use PHPUnit_Framework_TestCase;
+
+/**
+ * @covers \eLife\HypothesisClient\Credentials\JWTSigningCredentials
+ */
+class JWTSigningCredentialsTest extends PHPUnit_Framework_TestCase
+{
+    /**
+     * @test
+     */
+    public function it_will_have_a_start_time()
+    {
+        $credentials = new JWTSigningCredentials('foo', 'baz', 'authority', new Clock());
+        $this->assertGreaterThanOrEqual(time(), $credentials->getStartTime());
+    }
+
+    /**
+     * @test
+     */
+    public function it_may_have_an_expire_time()
+    {
+        $credentials = new JWTSigningCredentials('foo', 'baz', 'authority', new Clock());
+        $this->assertGreaterThan(0, $credentials->getExpire());
+        $credentials = new JWTSigningCredentials('foo', 'baz', 'authority', new Clock(), 100);
+        $this->assertEquals(100, $credentials->getExpire());
+    }
+
+    /**
+     * @test
+     */
+    public function it_can_generate_a_jwt_token()
```

Comment:
this is the only test that this class needs imho: one public method with no conditionals or loops, one functionality to cover